### PR TITLE
styled-components stylelint

### PR DESCRIPTION
**Pull request by kyleknighted**
## Description

Adds linting to styled-components 

## Motivation and Context

Need ability to quickly find errors in new styles
Able to run `yarn lint:styles` or `yarn lint:styles:path <path>`

## How Has This Been Tested?

running `yarn lint:styles` or `yarn lint:changes`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/dev/lintChanges.sh [hunk 1]
```diff
@@ -6,4 +6,6 @@
 # It will give you an idea of the ci result, before pushing your changes.
 # You may need to set the correct permissions for this file by running `chmod +x ./dev/lintChanges.sh`
 
-git diff --name-only origin/master...|grep -E '^graylog2-web-interface/(.*).js(x)?$'|sed s,graylog2-web-interface/,,g|xargs yarn lint:path
\ No newline at end of file
+git diff --name-only origin/master...|grep -E '^graylog2-web-interface/(.*).js(x)?$'|sed s,graylog2-web-interface/,,g|xargs yarn lint:path
+
+git diff --name-only origin/master...|grep -E '^graylog2-web-interface/(.*).js(x)?$'|sed s,graylog2-web-interface/,,g|xargs yarn lint:styles
```
Comment (edmundoa) [inline graylog2-web-interface/dev/lintChanges.sh]: I guess we'll want to do a similar check in Jenkins and include its results along with ESLint checks. It looks like we need an external formatter for that: [stylelint-checkstyle-formatter](https://github.com/davidtheclark/stylelint-checkstyle-formatter).
The script running in Jenkins is stored in Jenkins itself, but I think it's time to move it to the repo to have versioning and make it harder to change it accidentally. If you feel like it you can take care of that here as well, or just leave it for another PR.

File: graylog2-web-interface/package.json [hunk 2]
```diff
@@ -19,6 +19,8 @@
     "build:analyze": "yarn build:stats && webpack-bundle-analyzer ./webpack/stats-$(git rev-parse --abbrev-ref HEAD).json ./target/web/build --mode=\"static\" --report=\"./webpack/analysis-$(git rev-parse --abbrev-ref HEAD).html\"",
     "lint": "eslint --ext js,jsx src",
     "lint:path": "eslint --ext js,jsx",
+    "lint:styles":"stylelint --",
+    "lint:styles:all":"stylelint './src/**/*.{js,jsx,tsx}'",
```
[the remark this answers is not in the document]
Comment (kyleknighted) [reply graylog2-web-interface/package.json]: I agree, but maybe we add CSS in as a secondary PR to reduce complexity?

[the remark this answers is not in the document]
Comment (edmundoa) [reply graylog2-web-interface/package.json]: Sounds good to me 👍

File: graylog2-web-interface/package.json [hunk 3]
```diff
@@ -19,6 +19,8 @@
     "build:analyze": "yarn build:stats && webpack-bundle-analyzer ./webpack/stats-$(git rev-parse --abbrev-ref HEAD).json ./target/web/build --mode=\"static\" --report=\"./webpack/analysis-$(git rev-parse --abbrev-ref HEAD).html\"",
     "lint": "eslint --ext js,jsx src",
     "lint:path": "eslint --ext js,jsx",
+    "lint:styles": "stylelint './src/**/*.{css,js,jsx,tsx}'",
```
Comment (edmundoa) [inline graylog2-web-interface/package.json]: One of the last commits added checks on `css` but results don't seem right, all I get when running the command is ` 1    ✖  Unexpected token (1:0)` for each `css` file.
In my opinion we should either look why that happens and ensure it works, or leave `css` files out of the styling for now and take care of them in a separate PR, as we discussed some days ago.

Comment (kyleknighted) [reply graylog2-web-interface/package.json]: Will leave out as previously discussed and we can review this issue in the future (if any CSS files exist) 😁

**Pull request merged**